core/sequencer/server.go

```diff
@@ -92,7 +102,34 @@ func NewServer(
 		tlog:      tlog,
 		tmap:      tmap,
 		mutations: mutations,
+		queue:     queue,
+	}
+}
+
+// RunBatch reads mutations out of the queue and calls CreateEpoch.
+func (s *Server) RunBatch(ctx context.Context, in *spb.RunBatchRequest) (*empty.Empty, error) {
+	batch, err := s.queue.ReadQueue(ctx, in.DomainId, in.MaxBatch)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "ReadQueue(): %v", err)
 	}
+	if int32(len(batch)) < in.MinBatch {
+		return &empty.Empty{}, nil
+	}
+	if err := s.queue.DeleteMessages(ctx, in.DomainId, batch); err != nil {
```

Comment:
What happens if anything goes wrong after this point?  Is there a recovery mechanism?  (Maybe add a comment describing failure modes).

Also, I'd kind have expected the `DeleteMessages` to happen at the bottom iff no errors were encountered?

Reply:
I can move delete to the bottom. 

Items in the queue are permitted to be dropped on the floor (not ideal), but the clients will retry.